src/zenml/cli/base.py

```diff
@@ -318,12 +324,22 @@ def clean(ctx: click.Context, yes: bool = False, local: bool = False) -> None:
         local: If you want to delete local files associated with the active
             stack.
     """
-    ctx.invoke(
-        down,
-    )
+    ctx.invoke(down)
+
     if local:
+        curr_version = version.parse(zenml_version)
+
+        global_version = GlobalConfiguration().version
+        if global_version is not None:
+            config_version = version.parse(global_version)
+
+            if config_version > curr_version:
+                error(
+                    "Due to this version mismatch, ZenML can not detect and "
+                    "shut down any running dashboards or clean any resources "
+                    "related to the active stack."
+                )
         _delete_local_files(force_delete=yes)
-        return
```

Comment:
If I remember correctly, the `local` clean is only supposed to clean the local artifacts stored for the active stack and nothing else. You definitely need to keep this `return`.